src/components/MAPI/apps/AppList/utils.tsx

```diff
@@ -0,0 +1,327 @@
+  catalog: compareAppCatalogIndexAppsByAppCatalog,
+  latest: compareAppCatalogIndexAppsByLatest,
+};
+
+/**
+ * Remove the `Giant Swarm` prefix from
+ * internal catalogs, to ease cognitive load.
+ * @param catalog
+ */
+export function computeAppCatalogUITitle(
+  catalog: applicationv1alpha1.IAppCatalog
+): string {
+  const prefix = 'Giant Swarm ';
+
+  if (
+    !isAppCatalogVisibleToUsers(catalog) &&
+    catalog.spec.title.startsWith(prefix)
+  ) {
+    return catalog.spec.title.slice(prefix.length);
+  }
+
+  return catalog.spec.title;
+}
+
+/**
+ * Sort catalogs in an alphabetical order, but group
+ * managed catalogs together at the top.
+ * @param a
+ * @param b
+ */
+function compareAppCatalogs(
+  a: applicationv1alpha1.IAppCatalog,
+  b: applicationv1alpha1.IAppCatalog
+) {
+  const aIsVisible = isAppCatalogVisibleToUsers(a);
+  const bIsVisible = isAppCatalogVisibleToUsers(b);
+
+  if (aIsVisible && !bIsVisible) {
+    return -1;
+  } else if (!aIsVisible && bIsVisible) {
+    return 1;
+  }
+
+  const aTitle = computeAppCatalogUITitle(a);
+  const bTitle = computeAppCatalogUITitle(b);
+
+  if (aTitle < bTitle) {
+    return -1;
+  } else if (aTitle > bTitle) {
+    return 1;
+  }
+
+  return 0;
+}
+
+/**
+ * Transform the app catalogs data structures into
+ * ones that can be used by the catalog sidebar.
+ * @param appCatalogs
+ * @param selectedAppCatalogs - The catalogs that are currently enabled in the sidebar.
+ * @param appCatalogErrors - Errors that occured during loading each catalog.
+ */
+export function mapAppCatalogsToFacets(
+  appCatalogs: applicationv1alpha1.IAppCatalog[] = [],
+  selectedAppCatalogs: Record<string, boolean> = {},
+  appCatalogErrors: Record<string, string> = {}
+): IFacetOption[] {
+  return appCatalogs.sort(compareAppCatalogs).map((appCatalog) => {
+    const uiTitle = computeAppCatalogUITitle(appCatalog);
+
+    return {
+      value: appCatalog.metadata.name,
+      checked: selectedAppCatalogs.hasOwnProperty(appCatalog.metadata.name),
+      label: (
+        <CatalogLabel
+          catalogName={uiTitle}
+          iconUrl={appCatalog.spec.logoURL}
+          description={appCatalog.spec.description}
+          error={appCatalogErrors[appCatalog.metadata.name]}
+        />
+      ),
+    };
+  });
+}
+
+/**
+ * Search through the apps and find one that is in the
+ * selected catalogs, and that matches the search query.
+ * @param searchQuery
+ * @param indexApps
+ */
+export function filterAppCatalogIndexApps(
+  searchQuery: string,
+  indexApps: IAppCatalogIndexApp[],
+  selectedAppCatalogs: Record<string, boolean>
+) {
+  const normalizedQuery = searchQuery.trim().toLowerCase();
+
+  return indexApps.filter((indexApp) => {
+    if (!selectedAppCatalogs.hasOwnProperty(indexApp.catalogName)) return false;
+
+    if (normalizedQuery.length < 1) return true;
+
+    if (indexApp.versions.length < 1) return false;
+    const version = indexApp.versions[0];
```

Comment:
This is slightly different from the GS API implementation: Here we only search in the latest version, not in all the versions, to improve performance.